pkg/resources/password_policy_acceptance_test.go

```diff
@@ -1,49 +1,64 @@
 )
 
 func TestAcc_PasswordPolicy(t *testing.T) {
 	accName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	m := func(minLength int, maxLength int, comment string) map[string]config.Variable {
+		return map[string]config.Variable{
+			"name":       config.StringVariable(accName),
+			"database":   config.StringVariable(acc.TestDatabaseName),
+			"schema":     config.StringVariable(acc.TestSchemaName),
+			"min_length": config.IntegerVariable(minLength),
+			"max_length": config.IntegerVariable(maxLength),
+			"comment":    config.StringVariable(comment),
+		}
+	}
 
-	resource.ParallelTest(t, resource.TestCase{
-		Providers:    acc.TestAccProviders(),
-		PreCheck:     func() { acc.TestAccPreCheck(t) },
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
 		CheckDestroy: nil,
 		Steps: []resource.TestStep{
 			{
-				Config: passwordPolicyConfig(accName, 10, 30, "this is a test resource", acc.TestDatabaseName, acc.TestSchemaName),
+				ConfigDirectory: config.TestNameDirectory(),
+				ConfigVariables: m(10, 30, "this is a test resource"),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("snowflake_password_policy.pa", "name", accName),
 					resource.TestCheckResourceAttr("snowflake_password_policy.pa", "min_length", "10"),
 					resource.TestCheckResourceAttr("snowflake_password_policy.pa", "max_length", "30"),
 				),
 			},
 			{
-				Config: passwordPolicyConfig(accName, 20, 50, "this is a test resource", acc.TestDatabaseName, acc.TestSchemaName),
+				ConfigDirectory: config.TestNameDirectory(),
+				ConfigVariables: m(20, 50, "this is a test resource"),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("snowflake_password_policy.pa", "min_length", "20"),
 					resource.TestCheckResourceAttr("snowflake_password_policy.pa", "max_length", "50"),
 				),
 			},
-			/*
-					todo [SNOW-928909]: fix once comments are working again for password policies
-					query CREATE PASSWORD POLICY IF NOT EXISTS "T_Kn1bY6?2kx"."}k*3DrsXP:w9TRK#4wtS"."9ec016f6-ce74-0c94-2bd5-dc46547dbeff" PASSWORD_MIN_LENGTH = 10 PASSWORD_MAX_LENGTH = 20 PASSWORD_MIN_UPPER_CASE_CHARS = 5 COMMENT = 'test comment' err 001420 (22023): SQL compilation error: invalid property 'COMMENT' for 'PASSWORD_POLICY'
-				{
-					Config: passwordPolicyConfig(accName, 20, 50, ""),
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr("snowflake_password_policy.pa", "comment", ""),
-					),
-				},
-			*/
 			{
+				ConfigDirectory: config.TestNameDirectory(),
+				ConfigVariables: m(20, 50, ""),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_password_policy.pa", "comment", ""),
```

Comment:
maybe you should also check other params in the previous test case, e.g. to show that when the comment is set the attribute is also set

Reply:
oky